libraries/botbuilder-dialogs-adaptive/src/adaptiveComponentRegistration.ts

```diff
@@ -0,0 +1,263 @@
+/**
+ * @module botbuilder-dialogs-adaptive
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ComponentRegistration } from 'botbuilder-core';
+import {
+    ResourceExplorer,
+    ComponentDeclarativeTypes,
+    DeclarativeType,
+    CustomDeserializer,
+} from 'botbuilder-dialogs-declarative';
+import { AdaptiveDialog } from './adaptiveDialog';
+import {
+    BeginDialog,
+    BeginSkill,
+    BreakLoop,
+    CancelAllDialogs,
+    CancelDialog,
+    ContinueLoop,
+    DeleteActivity,
+    DeleteProperties,
+    DeleteProperty,
+    DynamicBeginDialog,
+    EditActions,
+    EditArray,
+    EmitEvent,
+    EndDialog,
+    EndTurn,
+    ForEach,
+    ForEachPage,
+    GetActivityMembers,
+    GetConversationMembers,
+    GotoAction,
+    HttpRequest,
+    IfCondition,
+    LogAction,
+    RepeatDialog,
+    ReplaceDialog,
+    SendActivity,
+    SetProperties,
+    SetProperty,
+    SignOutUser,
+    SwitchCondition,
+    TelemetryTrackEventAction,
+    TraceActivity,
+    UpdateActivity,
+} from './actions';
+import {
+    OnActivity,
+    OnAssignEntity,
+    OnBeginDialog,
+    OnCancelDialog,
+    OnChooseEntity,
+    OnChooseIntent,
+    OnChooseProperty,
+    OnCondition,
+    OnConversationUpdateActivity,
+    OnDialogEvent,
+    OnEndOfActions,
+    OnEndOfConversationActivity,
+    OnError,
+    OnEventActivity,
+    OnHandoffActivity,
+    OnIntent,
+    OnInvokeActivity,
+    OnMessageActivity,
+    OnMessageDeleteActivity,
+    OnMessageReactionActivity,
+    OnMessageUpdateActivity,
+    OnQnAMatch,
+    OnRepromptDialog,
+    OnTypingActivity,
+    OnUnknownIntent,
+} from './conditions';
+import {
+    Ask,
+    AttachmentInput,
+    ChoiceInput,
+    ConfirmInput,
+    DateTimeInput,
+    NumberInput,
+    OAuthInput,
+    TextInput,
+} from './input';
+import {
+    AgeEntityRecognizer,
+    ConfirmationEntityRecognizer,
+    CrossTrainedRecognizerSet,
+    CurrencyEntityRecognizer,
+    DateTimeEntityRecognizer,
+    DimensionEntityRecognizer,
+    EmailEntityRecognizer,
+    GuidEntityRecognizer,
+    HashtagEntityRecognizer,
+    IpEntityRecognizer,
+    MentionEntityRecognizer,
+    MultiLanguageRecognizer,
+    NumberEntityRecognizer,
+    OrdinalEntityRecognizer,
+    PercentageEntityRecognizer,
+    PhoneNumberEntityRecognizer,
+    RecognizerSet,
+    RegexEntityRecognizer,
+    RegexRecognizer,
+    TemperatureEntityRecognizer,
+    UrlEntityRecognizer,
+} from './recognizers';
+import { LuisAdaptiveRecognizer } from './luis';
+import { QnAMakerRecognizer } from './qnaMaker';
+import { ResourceMultiLanguageGenerator, TemplateEngineLanguageGenerator } from './generators';
+import { ConditionalSelector, FirstSelector, MostSpecificSelector, RandomSelector, TrueSelector } from './selectors';
+import { CustomDialogLoader } from './customDialogLoader';
+
+type Type = {
+    $kind: string;
+    new (): unknown;
```

Comment:
I think a better constructor type is `new (...args: unknown[]): unknown` in case the type does support constructor parameters.